pkg/graph/resolvers/key_contact_category.go

```diff
@@ -0,0 +1,179 @@
+	}
+
+	// Apply changes
+	if categoryValue, exists := changes["category"]; exists && categoryValue != nil {
+		var category string
+		var ok bool
+
+		// Handle both string and *string types (GraphQL may pass either)
+		if str, isString := categoryValue.(string); isString {
+			category = str
+			ok = true
+		} else if strPtr, isStringPtr := categoryValue.(*string); isStringPtr && strPtr != nil {
+			category = *strPtr
+			ok = true
+		}
+
+		if !ok {
+			return nil, fmt.Errorf("category must be a string, got type %T", categoryValue)
+		}
+
+		// Sanitize and validate category name
+		sanitizedCategory := sanitization.SanitizeString(category)
+		if sanitizedCategory == "" {
+			return nil, fmt.Errorf("category name cannot be empty or whitespace-only")
+		}
+		existingCategory.Category = sanitizedCategory
+	}
+
+	// Set modified by
+	existingCategory.ModifiedBy = &principalAccount.ID
+
+	updatedCategory, err := sqlutils.WithTransaction(store, func(tx *sqlx.Tx) (*models.KeyContactCategory, error) {
+		updatedCategory, err := storage.KeyContactCategoryUpdate(tx, logger, existingCategory)
+		if err != nil {
+			return nil, fmt.Errorf("failed to update category with id %s: %w", id, err)
+		}
+		return updatedCategory, nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return updatedCategory, nil
+}
+
+// DeleteKeyContactCategory deletes a key contact category by its ID.
+// Returns the deleted category or an error.
+func DeleteKeyContactCategory(logger *zap.Logger, principal authentication.Principal, store *storage.Store,
+	id uuid.UUID,
+) (*models.KeyContactCategory, error) {
+	principalAccount := principal.Account()
+	if principalAccount == nil {
+		return nil, fmt.Errorf("principal doesn't have an account, username %s", principal.String())
+	}
+
+	// Use a transaction for delete (for audit triggers, etc.)
+	returnedCategory, err := sqlutils.WithTransaction(store, func(tx *sqlx.Tx) (*models.KeyContactCategory, error) {
+		returnedCategory, err := storage.KeyContactCategoryDelete(tx, logger, id)
+		if err != nil {
+			return nil, fmt.Errorf("unable to delete category. Err %w", err)
+		}
+
+		return returnedCategory, nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return returnedCategory, nil
+}
+
+// GetKeyContactCategory retrieves a key contact category by its ID.
+// Returns the category if found, or an error if not found or on failure.
+func GetKeyContactCategory(logger *zap.Logger, principal authentication.Principal, store *storage.Store,
+	id uuid.UUID,
+) (*models.KeyContactCategory, error) {
+	principalAccount := principal.Account()
+	if principalAccount == nil {
+		return nil, fmt.Errorf("principal doesn't have an account, username %s", principal.String())
+	}
+
+	category, err := storage.KeyContactCategoryGetByID(store, logger, id)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get category with id %s: %w", id, err)
+	}
+
+	if category == nil {
+		return nil, fmt.Errorf("category with id %s not found", id)
```

Comment:
This nil check is redundant since the storage function already returns an error when a category is not found